supervision/utils/file.py

```diff
@@ -1,10 +1,121 @@
+import csv
 import json
 from pathlib import Path
-from typing import List, Optional, Union
+from typing import Any, Dict, List, Optional, Union
 
 import numpy as np
 import yaml
 
+from supervision.detection.core import Detections
+
+
+class CSVSink:
+    """
+    A utility class for saving detection data to a CSV file. This class is designed to
+    efficiently serialize detection objects into a CSV format, allowing for the inclusion of
+    bounding box coordinates and additional attributes like confidence, class ID, and tracker ID.
+
+    The class supports the capability to include custom data alongside the detection fields,
+    providing flexibility for logging various types of information.
+
+    Args:
+        filename (str): The name of the CSV file where the detections will be stored.
+                        Defaults to 'output.csv'.
+
+    Usage:
+        ```python
+        from supervision.utils.detections import Detections
```

Comment:
I actually developed the demo using a track object algorithm. [here](https://colab.research.google.com/drive/1DyKR_euGxHy4YXD_QGA6bwsBNDs9mScV?usp=sharing) 

Do you think an example like this could work? 
```python
        import numpy as np
        import supervision as sv
        from ultralytics import YOLO
        import time

        model = YOLO("yolov8n.pt")
        tracker = sv.ByteTrack()
        box_annotator = sv.BoundingBoxAnnotator()
        label_annotator = sv.LabelAnnotator()
        csv_sink = sv.CSVSink(...)

        def callback(frame: np.ndarray, _: int) -> np.ndarray:
            start_time = time.time()
            results = model(frame)[0]
            detections = sv.Detections.from_ultralytics(results)
            detections = tracker.update_with_detections(detections)

            labels = [
                f"#{tracker_id} {results.names[class_id]}"
                for class_id, tracker_id
                in zip(detections.class_id, detections.tracker_id)
            ]
            time_frame = (time.time() - start_time)

            csv_sink.append(detections, custom_data={"processing_time": time_frame})

            annotated_frame = box_annotator.annotate(
                frame.copy(), detections=detections)
            return label_annotator.annotate(
                annotated_frame, detections=detections, labels=labels)

        csv_sink.open()
        sv.process_video(
            source_path="people-walking.mp4",
            target_path="result.mp4",
            callback=callback
        )
        csv_sink.close()

``` 

I added an example of time spent processing a frame, but we can use any custom value.

Reply:
I think the example in the docs " reference " section should be relatively minimal. We should not introduce elements like annotators in the CSV sink section of docs.

But we should create additional, more comprehensive tutorials showing how to combo those independent tools. But this is outside of the scope.